compiler/rustc_ast/src/ast.rs

```diff
@@ -1563,7 +1574,23 @@ impl MacArgs {
         match self {
             MacArgs::Empty => TokenStream::default(),
             MacArgs::Delimited(.., tokens) => tokens.clone(),
-            MacArgs::Eq(.., token) => TokenTree::Token(token.clone()).into(),
+            MacArgs::Eq(_, MacArgsEqKind::Ast(expr)) => {
```

Comment:
>I didn't use `nt_to_tokenstream` in `MacArgs::inner_tokens` because (a) it's not currently necessary, and (b) it's a bit of a pain to do so, because `rustc_ast` can't easily use code from `rustc_parse`.

What are the cases requiring `inner_tokens` for `MacArgs::Eq` in `rustc_ast`?
AFAK, the tokens are only needed in `rustc_expand` and `rustc_ast_lowering`, everywhere else the `ast::Expr` can be used directly.
I suggest panicking here.

Reply:
The `Ast` case occurs in one ui test: `src/test/ui/proc-macro/proc-macro-gates.rs`, for this code:
```rust
#[empty_attr = "y"] //~ ERROR: key-value macro attributes are not supported
fn _test3() {}
```
Here's a partial stack trace:
```
  17:     0x7fb9129b67b3 - <rustc_ast[eca82849642287e8]::ast::MacArgs>::inner_tokens
                               at /home/njn/dev/rust2/compiler/rustc_ast/src/ast.rs:1593:17
  18:     0x7fb911ec2e48 - <rustc_expand[4e78a34ef71c6af4]::expand::MacroExpander>::expand_invoc
                               at /home/njn/dev/rust2/compiler/rustc_expand/src/expand.rs:692:40
  19:     0x7fb911ec2e48 - <rustc_expand[4e78a34ef71c6af4]::expand::MacroExpander>::fully_expand_fragment
                               at /home/njn/dev/rust2/compiler/rustc_expand/src/expand.rs:451:62
  20:     0x7fb911ec0b37 - <rustc_expand[4e78a34ef71c6af4]::expand::MacroExpander>::expand_crate
                               at /home/njn/dev/rust2/compiler/rustc_expand/src/expand.rs:378:21
  21:     0x7fb90d562af4 - rustc_interface[50ab7c1b72438f3d]::passes::configure_and_expand::{closure#1}::{closure#1}
                               at /home/njn/dev/rust2/compiler/rustc_interface/src/passes.rs:343:50
```

Reply:
Ok, let's keep this as is, I'll look at this myself later.